### DISCUSSION TITLE
Is it possible to control my heating with ebusd or not ?

### DISCUSSION THREAD

**Discussion by chivagio** [john30/ebusd#1083] [Q&A] [answered]
Hi everyone,

**_TLDR: is it possible or not to control my heating with ebusd ?_**

I'm writing here in a final act of desperation. I've looked anywhere on the web and this is my last option before eventually returning the eBUS adapter. Anyway, here goes : 

I have a Vaillant ecoTEC pro VCW 286/5-3 that is controlled by a VRT 250 thermostat. I want to be able to turn on/off the heating within Home Assistant based on various temperatures around the house. For this I've ordered an eBUS adapter and managed to integrate it in HA using the MQTT discovery. I can see lots of sensors and two writable ones.

I may be stupid, but I have absolutely NO CLUE how to turn the heating on. All I can see is various informative sensors. I've tried "_ebusctl find -w_ and I see a bunch of commands but "SetMode" or "SetModeOverride" **does not exist**. (and even if it did exist, I have no idea about the syntax or what data I should send to turn the heating on)

So is there a way to turn the heating on/off like this or not? Is ebusd meant for different use cases?

Thanks!

**Comment by andrewt27** [comment]
The VRT 250 controller you have is a regular thermostat and works on the principle of turning the boiler on/off, and there may be a conflict here. Because the boiler will always respond to the thermostat.
You would have to change the boiler controller to one that communicates with the boiler via BUS, e.g. multiMATIC VRC 700

**Reply by chivagio** [reply]
Well I'm actually trying to get rid of the controller altogether. All I want is to be able to control the heating myself when needed (via HA)

**Reply by andrewt27** [reply]
Yes but boiler needs to have temperature readings if it is controlled by BUS controller

**Reply by chivagio** [reply]
isn't there any way of simulating the temp readings, demanding heat when needed ?

**Reply by andrewt27** [reply, marked as answer]
I'm afraid not

**Reply by Dinth** [reply]
Im pretty sure that i have seen a long thread regarding the possibility of simulating the controller using ebusd somewhere and as far as i remember it should be technically possible, but replacing the controller is not something ebusd is for.

**Reply by vmalaga** [reply]
yes you can use this command, need to change f35 with the name of your circuit, and change at the csv definition of your controller, the `r` to `r;w` at the `RoomTemp` definition

`ebusctl write -c f35 RoomTemp "21;ok"`

`r;w,,RoomTemp,RaumtemperaturIst,,,,"0000",,,tempsensor,,,raw value current room temperature and sensorstate (without any correction)`

**Reply by takui1** [reply]
Hi, thanks for the suggestion. I just want to be sure I understand this correctly before trying it on my Vaillant system.

I asked ChatGPT about this, and the answer I got is that the CSV definition in ebusd acts only as a “mapping file”. In other words, it tells ebusd how to interpret a command, but it does not enable or unlock new commands that the controller’s firmware does not actually support.
So even if I change the RoomTemp entry from ro to r;w in the CSV, ebusd may allow me to attempt a write, but the Vaillant controller will simply ignore it or return an error if the firmware doesn’t implement that write command.

From what I understand, your example with circuit F35 works because that specific controller actually supports RoomTemp as a writable value at firmware level. Vaillant controllers (VRC700 / HK1 / HK2), on the other hand, seem to expose RoomTemperature as read-only in every official Vaillant CSV, meaning the firmware does not accept overrides of the current room temperature.

Can you confirm whether the Vaillant system truly accepts this write command, or does this method only work on devices where the underlying firmware exposes RoomTemp as writable (like F35)?

**Reply by JonesGW-MGD** [reply]
Typically you can emulate the SetMode command that controllers use.  If you search these forums for ```SetModeOverride```  you can see the work that others have done but essentially it's a clone of the SetMode command.  You might want to start here and work backwards: https://github.com/john30/ebusd/discussions/1492 

This way you can control flowtemperatures and adjust when whatever you are planning to use to monitor temperature reaches the desired state.

Once you have gained a fuller understanding you might want to raise a new discussion item and share your set-up and devices - because you'll appreciate it's a bit hard to generalise over such a wide range of Vaillant heat sources such as boilers, air exchangers, and heat pumps.

Of course, you're now venturing into a domain that I suspect very few people have ventured (including me) - replacing or emulating a controller using ebusd and a domain for which the application wasn't designed.   Heat sources are both expensive and dangerous!
